developer_manual/digging_deeper/task_processing.rst

```diff
@@ -0,0 +1,346 @@
+
+If you would like to use the text processing functionality in a client, there are also OCS endpoints available for this: :ref:`OCS Text Processing API<ocs-taskprocessing-api>`
+
+Tasks types
+^^^^^^^^^^^
+The following built-in task types are available:
+
+ * ``'core:text2text'``: This task allows passing an arbitrary prompt to the language model. It is implemented by ``\OCP\TaskProcessing\TaskTypes\TextToText``
+    * Input shape:
+       * ``input``: ``Text``
+    * Output shape:
+       * ``output``: ``Text``
+ * ``'core:text2text:headline'``: This task will generate a headline for the passed input text. It is implemented by ``\OCP\TaskProcessing\TaskTypes\TextToTextHeadline``
+     * Input shape:
+        * ``input``: ``Text``
+     * Output shape:
+        * ``output``: ``Text``
+ * ``'core:text2text:topics'``: This task will generate a comma-separated list of topics for the passed input text. It is implemented by ``\OCP\TaskProcessing\TaskTypes\TextToTextTopics``
+      * Input shape:
+         * ``input``: ``Text``
+      * Output shape:
+         * ``output``: ``Text``
+ * ``'core:text2text:summary'``: This task will summarize the passed input text. It is implemented by ``\OCP\TaskProcessing\TaskTypes\TextToTextSummary``
+      * Input shape:
+        * ``input``: ``Text``
+      * Output shape:
+         * ``output``: ``Text``
+ * ``'core:audio2text'``: This task type is for transcribing audio to text. It is implemented by ``\OCP\TaskProcessing\TaskTypes\AudioToText``
+     * Input shape:
+        * ``input``: ``Audio``
+     * Output shape:
+        * ``output``: ``Text``
+ * ``'core:text2image'``: This task type is for generating images from text prompts. It is implemented by ``\OCP\TaskProcessing\TaskTypes\TextToImage``
+      * Input shape:
+         * ``input``: ``Text``
+         * ``numberOfImages``: ``Number``
+      * Output shape:
+         * ``output``: ``ListOfImages``
+
+Input and output shapes
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Each task type defines how its input and output should look. This is called the input and output shape.
+
+For example the TextToImage type defines its input shape as follows:
+
+.. code-block:: php
+
+    /**
+     * @return ShapeDescriptor[]
+     * @since 30.0.0
+     */
+    public function getInputShape(): array {
+        return [
+            'input' => new ShapeDescriptor(
+                $this->l->t('Prompt'),
+                $this->l->t('Describe the image you want to generate'),
+                EShapeType::Text
+            ),
+            'numberOfImages' => new ShapeDescriptor(
+                $this->l->t('Number of images'),
+                $this->l->t('How many images to generate'),
+                EShapeType::Number
+            ),
+        ];
+    }
+
+The task input and output is always an array. In this case, the task input for TextToImage must have an array key named ``'input'`` which must contain a text and an array key named ``'numberOfImages'`` which must contain a number.
```

Comment:
The task input and output **are** always represented as arrays.

Reply:
```suggestion
The task input and output are always represented by an associative array. In this case, the task input for TextToImage must have an array key named ``'input'`` which must contain a text and an array key named ``'numberOfImages'`` which must contain a number.
```